dist/GetFullComputerName.nsh

```diff
@@ -1,19 +1,16 @@
 !macro _GetFullComputerName
-
-  Var /GLOBAL ComputerName
-  Var /GLOBAL PrimaryDomainName
-  Var /GLOBAL FullComputerName
+  Push $R0
+  Push $R1
 
   ReadRegStr $0 HKLM "System\CurrentControlSet\Control\ComputerName\ActiveComputerName" "ComputerName"
-  StrCpy $ComputerName $0
+  StrCpy $R0 $0
 
   StrCpy $0 0
   EnumRegKey $1 HKLM "System\CurrentControlSet\Services\Tcpip\Parameters\DNSRegisteredAdapters" $0
   ReadRegStr $0 HKLM "System\CurrentControlSet\Services\Tcpip\Parameters\DNSRegisteredAdapters\$1" "PrimaryDomainName"
 
-  StrCpy $PrimaryDomainName $0
-  StrCpy $FullComputerName "$ComputerName.$PrimaryDomainName"
+  StrCpy $R1 "$R0.$0"
 
```

Comment:
The concatenation should only happen when both names are actually set. Otherwise we end up in a name like `WINDOWS.` with a dot at the end.